processor/spanmetricsprocessor/processor.go

```diff
@@ -66,24 +67,32 @@ type processorImp struct {
 	nextConsumer    consumer.Traces
 
 	// Additional dimensions to add to metrics.
-	dimensions []Dimension
+	dimensions []KeyValuePair
+
+	// Additional resourceAttributes to add to metrics.
+	resourceAttributes []KeyValuePair
 
 	// The starting time of the data points.
 	startTime time.Time
 
 	// Call & Error counts.
-	callSum map[metricKey]int64
+	callSum map[resourceKey]map[metricKey]int64
 
 	// Latency histogram.
-	latencyCount         map[metricKey]uint64
-	latencySum           map[metricKey]float64
-	latencyBucketCounts  map[metricKey][]uint64
+	latencyCount         map[resourceKey]map[metricKey]uint64
+	latencySum           map[resourceKey]map[metricKey]float64
+	latencyBucketCounts  map[resourceKey]map[metricKey][]uint64
 	latencyBounds        []float64
-	latencyExemplarsData map[metricKey][]exemplarData
+	latencyExemplarsData map[resourceKey]map[metricKey][]exemplarData
+
+	// List of seen resource attributes.
+	// Map structure for faster lookup.
+	resourceAttrList map[resourceKey]bool
 
-	// A cache of dimension key-value maps keyed by a unique identifier formed by a concatenation of its values:
+	// A cache of dimension and resource attribute key-value maps keyed by a unique identifier formed by a concatenation of its values:
```

Comment:
it's just the cache of the dimensions but not resource attributes.

Reply:
I guess we forgot to reset the comment back after you split the changed cache into two

Reply:
fixed (separated into it's own comment)